components/findMatch.tsx

```diff
@@ -0,0 +1,112 @@
+import React, {Component} from "react";
+import { Button, TextField, ToggleButton, ToggleButtonGroup } from "@mui/material";
+
+export class FindMatch extends Component {
+
+    selected : boolean = false;
+    id : string = "";
+
+    changeStatus = () => {
+        this.selected = !this.selected;
+        this.updateView();
+    }
+
+    updateView = () => {
+        const el = document.getElementById("search");
+        
+        if(el){
+            el.innerHTML  = this.selected ? "Cancel" : "Find match";
+        }
+    }
+
+    makeRequest = (createGame : boolean) => {
+
+        //Send the first request to be added to the waiting list
+        const reqWaitingListOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ 
+                name : document.getElementById("playerName").value ,
+                waitingList : true, //Riguada una richista che ha a che fare con la waiting list 
+                accessWaitingList : createGame, // stabilisce se creare aggiungere o rimuovere dalla waiting list
+                id : this.id
+            })
+        };
+        
+        const url = "/api/game/findGame";
+        
+        fetch(url, reqWaitingListOptions)
+        .then(response => response.json())
+        .then(data => {
+            
+            if(createGame){
+                this.id = data.id;
```

Comment:
non sono sicuro che funzioni, il `this` di javascript è [molto strano](https://blog.bitsrc.io/what-is-this-in-javascript-3b03480514a7), sei sicuro di averlo provato e ricevi i valori giusti?.
Comunque con il react funzionale poi non ti dovrai fare questi problemi